app/workers/messaging.php

```diff
@@ -20,59 +33,197 @@
 
 class MessagingV1 extends Worker
 {
-    protected ?Adapter $sms = null;
-    protected ?string $from = null;
+    protected ?SMSAdapter $sms = null;
+    protected ?PushAdapter $push = null;
+    protected ?EmailAdapter $email = null;
+
+    protected ?Database $dbForProject = null;
+
 
     public function getName(): string
     {
-        return "mails";
+        return "messaging";
     }
 
+
     public function init(): void
     {
-        $dsn = new DSN(App::getEnv('_APP_SMS_PROVIDER'));
-        $user = $dsn->getUser();
-        $secret = $dsn->getPassword();
-
-        $this->sms = match ($dsn->getHost()) {
-            'mock' => new Mock($user, $secret), // used for tests
-            'twilio' => new Twilio($user, $secret),
-            'text-magic' => new TextMagic($user, $secret),
-            'telesign' => new Telesign($user, $secret),
-            'msg91' => new Msg91($user, $secret),
-            'vonage' => new Vonage($user, $secret),
-            default => null
-        };
-
-        $this->from = App::getEnv('_APP_SMS_FROM');
     }
 
     public function run(): void
     {
-        if (empty(App::getEnv('_APP_SMS_PROVIDER'))) {
-            Console::info('Skipped sms processing. No Phone provider has been set.');
-            return;
+        $project = new Document($this->args['project']);
+        $this->dbForProject = $this->getProjectDB($project);
+
+        $message = $this->dbForProject->getDocument('messages', $this->args['messageId']);
+
+        $provider = $this->dbForProject->getDocument('providers', $message->getAttribute('providerId'));
+
+        $this->processMessage($message, $provider);
+    }
+
+    private function processMessage(Document $messageRecord, Document $providerRecord): void
```

Comment:
Let's name these params message and provider

Reply:
It is a bit problematic as we need message and provider for Message type and Provider adapter type.

Reply:
We can rename $provider -> $adapter, $providerRecord -> $provider, $message -> $data and $messageRecord -> $message

Let's avoid using `record` in general, it introduces extra terminology for what we are already calling `document` elsewhere

Reply:
done this